sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProviderBuilder.java

```diff
@@ -52,6 +54,9 @@ public SdkMeterProviderBuilder setResource(@Nonnull Resource resource) {
    * @return An initialized TracerSdkFactory.
    */
   public SdkMeterProvider build() {
-    return new SdkMeterProvider(clock, resource);
+    Resource resourceWithFallbacks =
+        Resource.create(
+            FALLBACK_MANDATORY_ATTRIBUTES.toBuilder().putAll(resource.getAttributes()).build());
```

Comment:
This is how I initially proposed the spec PR, but not how I reworked it shortly after and how it got approved 😃 
Now the spec basically says that we should treat the service.name exactly the same as we already treat the telemetry.sdk.name: It's there by default, but if the user set's their own resource explicitly, they need to merge it in manually.

See https://github.com/open-telemetry/opentelemetry-specification/pull/1294#issuecomment-747589223 and https://github.com/open-telemetry/opentelemetry-specification/blob/39fa82f0a7e58975782520c304768d63da0c7612/specification/resource/sdk.md (not yet merged).

If you think the first version of the PR was better, you should say so on the spec. 

@anuraaga @jkwatson You approved the spec PR after I made this change, but maybe you want to check again?

Reply:
I removed my approval, because I was mostly going on the PR description, which seems to match what I've implemented here. What's now written in the spec is no longer comprehensible to me. :(

Reply:
@jkwatson To really dismiss your approval you have to expand the box at the bottom of the PR and click the dots next to your approval there (or add a request-changes review). Adding a comment-only review does not remove the approval.